docs/design-docs/reporting-sot.md

```diff
@@ -0,0 +1,533 @@
+# Design doc: Reporting source-of-truth (SoT) views
+
+This doc describes the design of a new set of reporting SQL views that aim
+to make the CCAO's reporting more consistent, durable, debuggable, and useful.
+
+## Motivation
+
+The Data Department gets asked to fill many ad-hoc reporting requests for
+various aggregate stats, usually partitioned by geography. This has resulted
+in a buildup of ad-hoc reporting views and tables over time. These ad-hoc views
+are difficult to debug and trust, because each of them is complicated in a
+slightly different way.
+
+Rather than continue to create such ad-hoc views, we should consolidate our
+reporting efforts into a few larger views that cover the majority of CCAO
+reporting needs. These new views will cover different groupings, geographies,
+and valuation stages, and will use a consistent methodology and schema.
+They will serve as the new "source-of-truth" for both internal and external
+CCAO reporting.
+
+## Requirements
+
+Each of the following requirements has two components: A motivation ("why do
+do we want to do this?") plus a list of requirements ("how will we do this?").
+
+### Durable / long-lived
+
+#### Motivation
+
+Since its inception in 2018, the CCAO's Data Department has created, used, and
+deprecated many reporting assets (views, tables, and extracts). This iteration
+was necessary due to changing systems and reporting requirements/capabilities.
+However, this constant change creates two problems:
+
+1. It's hard to know which reporting asset was used to create past output, as
+  the asset may be deprecated/gone.
+2. It's hard to know which asset to use for *current* reporting/diagnostic
+  needs, as the available assets are often changing.
+
+Now that the pace of institutional change has slowed a bit, it's possible to
+create a more permanent set of reporting views that would help solve these
+problems.
+
+#### Requirements
+
+We should be able to:
+
+- Reference columns or tables by name without fear of them changing or
+  disappearing in a few years
+- Store the data in a persistent and/or fully reproducible way, such that
+  there is a near-zero chance of data loss
```

Comment:
This becomes much more complex if the tables need to be immutable, see open questions section below.